aips/aip-124.md

```diff
@@ -0,0 +1,199 @@
+---
+aip:
+title: x-chain DAA authentication using Sign Personal Message with Sui
+author: 0xmaayan, igor-aptos
+Status: Draft
+type: Framework
+discussions-to (*optional):
+created: 06/11/2025
+---
+
+# AIP-X - x-chain DAA authentication using Sign Personal Message with Sui
+
+## Summary
+
+Implements authentication function giving Sui users native Aptos accounts,
+and allowing them to sign transaction on Aptos blockchain, directly from their Sui wallet.
+This is done using derivable account abstraction (AIP-113).
+
+## High-level Overview
+
+Allow Sui accounts (Sui Ed25519 key pairs) to create Aptos transactions,
+and be able to autheneticate and execute them directly on the Aptos blockchain, without interaction with Sui Blockchain.
+We do so by utilizing ["Sign Personal Message"](https://docs.sui.io/standards/wallet-standard#implementing-features) specification, and defining
+a message format that gives enough details to users so they understand what they are signing reasonably well.
+
+## Impact
+
+This will allow users with Sui Wallets to be able to interact with Aptos blockchain
+
+## Alternative Solutions
+
+Alternatively, wallets from other chains can implement integration with Aptos natively, providing more seamless experience to the users.
+
+## Specification and Implementation Details
+
+dApp is required to issue signPersoanlMessage request to the wallet with:
+
+```
+<domain> wants you to sign in with your Sui account:
+<sui_address>
+
+Please confirm you explicitly initiated this request from <domain>. You are approving to execute transaction <entry_function> on Aptos blockchain (<network_name>).
+
+Nonce: <aptos_txn_digest>
+```
+
+And submit transaction to Aptos with Authenticator:
+
+```
+AccountAuthenticator::Abstraction {
+    function_info: FunctionInfo::from("0x1::sui_derivable_account::authenticate"),
+    auth_data: AbstractionAuthData::DerivableV1 {
+        signing_message_digest: aptos_txn_digest,
+        abstract_signature: bcs::to_bytes(SuiAbstractSignature::SuiDerivedSignature {
+            signature: sign_personal_message_signature,
+        }),
+        abstract_public_key: bcs::to_bytes(SuiAbstractPublicKey {
+            sui_account_address,
+            domain,
+        }),
+    },
+}
+```
+
+with types:
+
+```
+enum SuiAbstractSignature has drop {
+  SuiDerivedSignature {
+      signature: vector<u8>,
+  },
+}
+
+struct SuiAbstractPublicKey has drop {
+  sui_account_address: vector<u8>,
+  domain: vector<u8>,
+}
+```
+
+And above message will be authorized with `0x1::sui_derivable_account::authenticate`,
+for aptos account with address derevied from function_info and abstract_public_key with account_abstraction::derive_account_address.
```

Comment:
derevied --> derived

it's nice that you are using `code blocks`, but you should use the everywhere: e.g., also for:

function_info
abstract_public_key
account_abstraction::derive_account_address